Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
def plot_energy_gibbs(f, x, ax):
    ax[1].plot(x, f(x), color='red')
    ax[1].set_xticks([])
    ax[1].set_yticks([])
    ax[1].set_title('Energy Function - look for min')

    ax[0].plot(x, -f(x), color='blue')
    ax[0].set_xticks([])
    ax[0].set_yticks([])
    ax[0].set_title('Probability Density Function - look for max')
    
    return ax

f = lambda x: x**4 + 1* x**3 - 10 * x**2 + 1
x = np.linspace(-4, 3.5, 100)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[4], line 15[0m
[1;32m     12[0m     [38;5;28;01mreturn[39;00m ax
[1;32m     14[0m f [38;5;241m=[39m [38;5;28;01mlambda[39;00m x: x[38;5;241m*[39m[38;5;241m*[39m[38;5;241m4[39m [38;5;241m+[39m [38;5;241m1[39m[38;5;241m*[39m x[38;5;241m*[39m[38;5;241m*[39m[38;5;241m3[39m [38;5;241m-[39m [38;5;241m10[39m [38;5;241m*[39m x[38;5;241m*[39m[38;5;241m*[39m[38;5;241m2[39m [38;5;241m+[39m [38;5;241m1[39m
[0;32m---> 15[0m x [38;5;241m=[39m [43mnp[49m[38;5;241m.[39mlinspace([38;5;241m-[39m[38;5;241m4[39m, [38;5;241m3.5[39m, [38;5;241m100[39m)

[0;31mNameError[0m: name 'np' is not defined

